libs/dh/admin/feature-permissions/src/lib/details/tabs/dh-admin-permission-roles.component.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CommonModule } from '@angular/common';
+import {
+  Component,
+  inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  SimpleChanges,
+} from '@angular/core';
+import { WattSpinnerModule } from '@energinet-datahub/watt/spinner';
+import { WattCardModule } from '@energinet-datahub/watt/card';
+import { WattTableColumnDef, WattTableDataSource, WATT_TABLE } from '@energinet-datahub/watt/table';
+import { WattEmptyStateModule } from '@energinet-datahub/watt/empty-state';
+import { TranslocoModule, TranslocoService } from '@ngneat/transloco';
+import { PermissionDto, graphql } from '@energinet-datahub/dh/shared/domain';
+import { Apollo, QueryRef } from 'apollo-angular';
+import { Subscription } from 'rxjs';
+import { ApolloError } from '@apollo/client';
+import { LetModule } from '@rx-angular/template/let';
+import type { ResultOf } from '@graphql-typed-document-node/core';
+
+type UserRole = ResultOf<
+  typeof graphql.GetPermissionDetailsDocument
+>['permission']['userRoles'][number];
+
+@Component({
+  selector: 'dh-admin-permission-roles',
+  templateUrl: './dh-admin-permission-roles.component.html',
+  styles: [``],
```

Comment:
question: some elements in the template have classes, but the styling is missing. Is it coming in a follow up PR?

Reply:
Yes will come in part 4 :-)